clang/include/clang/Driver/Options.td

```diff
@@ -798,6 +798,8 @@ def MD : Flag<["-"], "MD">, Group<M_Group>,
     HelpText<"Write a depfile containing user and system headers">;
 def MMD : Flag<["-"], "MMD">, Group<M_Group>,
     HelpText<"Write a depfile containing user headers">;
+def MMMD : Flag<["-"], "MMMD">, Group<M_Group>,
```

Comment:
I am wondering whether we could keep using the old compiler option `-MMD` for this new behavior, and use triple/language etc to choose the new behavior. The reason is that users usually use the existing options in their cmake file for dependency files.

Reply:
OK, I added the MMMD option primarily to avoid altering the existing user habits as much as possible. If everyone agrees that this behavior can be enabled by default, then I will remove this option.